server/openapi_metric.go

```diff
@@ -0,0 +1,42 @@
+package server
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	metricAPI "github.com/bytebase/bytebase/metric"
+	"github.com/bytebase/bytebase/plugin/metric"
+	"github.com/labstack/echo/v4"
+)
+
+func openAPIMetricMiddleware(s *Server, next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) (err error) {
+		start := time.Now()
+
+		defer func() {
+			duration := time.Since(start)
+			requestMethod := c.Request().Method
+			requestPath := c.Path()
+			responseCode := c.Response().Status
+
+			if s.MetricReporter != nil {
+				s.MetricReporter.report(&metric.Metric{
+					Name:  metricAPI.OpenAPIMetricName,
+					Value: 1,
+					Labels: map[string]string{
+						"duration":       fmt.Sprintf("%f", duration.Seconds()),
```

Comment:
I am not sure using seconds will yield meaningful results, it will be either 0 or 1 in most cases. Maybe call it latency_ms

Reply:
I changed to milliseconds now